### DISCUSSION TITLE
How can I create a one-column slide in Powerpoint via Jupyter and Quarto?

### DISCUSSION THREAD

**Discussion by NickCH-K** [quarto-dev/quarto-cli#5609] [Q&A]
I am trying to use Quarto to render a Jupyter notebook to Powerpoint slides. I would like to render one slide as a "Content with Caption" slide, which is the style intended for single-column slides with both text and non-text, as per the [Quarto docs on Powerpoint](https://quarto.org/docs/presentations/powerpoint.html).

How can I generate both text and a graph in the same column in Jupyter, in a way Quarto will recognize?

Using the default Pandoc Powerpoint template, I have tried:

```
(markdown chunk) text
(code chunk) graph
```

which renders as a two-column slide

```
(code chunk) print('my text as a string')
graph
```

which also renders as a two-column slide (with code formatting for the text now)

```
(markdown chunk)
:::: {.columns}

::: {.column width="100%"}
text
(code chunk)
graph
(markdown chunk)
:::

::::
```

which prints the text as intended in a single column but omits the graph entirely.

Is it possible to do this? Thank you. And yes, "use the Quarto editor in RStudio instead of using Jupyter" would be a great solution but is not an option available to me; I have to use Jupyter for this.

(this is a crosspost of a [StackOverflow question](https://stackoverflow.com/questions/75993009/getting-a-single-column-slide-when-using-quarto-to-render-a-jupyter-notebook-to) and a [Posit Community](https://community.rstudio.com/t/creating-a-one-column-slide-in-powerpoint-via-jupyter-and-quarto/164269/2) question, neither of which got any responses).

**Comment by cderv** [comment]
I think we have an issue here. I don't know exactly how Pandoc will "guess" which layout to use but it could be that the intermediate `.md` that we generate before Pandoc conversion contains element that will make it no possible to use the Content with Caption layout. 

This is the content of the intermediate .md BTW
````markdown
---
title: Quarto Basics
format: pptx
keep-md: true
---


## Polar Axis

For a demonstration of a line plot on a polar axis, see @fig-polar.

::: {.cell execution_count=1}

::: {.cell-output .cell-output-display}
![A line plot on a polar axis](test2_files/figure-pptx/fig-polar-output-1.png){#fig-polar}
:::
:::
````

We have some specific div that maybe are causing Pandoc to not choose the right layout 🤔 

i'll open an issue to track this.

**Reply by cderv** [reply]
Maybe not issue: See below: https://github.com/quarto-dev/quarto-cli/discussions/5609#discussioncomment-5966455

**Reply by TerrifiedJelly** [reply]
I'm having the exact same issue on R. I've read all the documentation which shows you need the slides in your template to be called Two Content, Comparison, Content with Caption etc but I can't find anywhere where it shows clear examples of how to select the slide you need to use. Please can we get some clearer examples?
At the moment, I'm feeding back to the business that this isn't usable for when we're attempting. I think it probably is, but the examples are too brief and after a full day working on just this one issue, I'm giving up! All I've found online is people asking this exact question. No one seems to have cracked this.

**Reply by mcanouil** [reply]
@TerrifiedJelly You don't select as explain in https://github.com/quarto-dev/quarto-cli/discussions/5609#discussioncomment-5966455, Pandoc guesses which layout to use.
The issue is not with Quarto here.

See https://github.com/jgm/pandoc/issues/5258 with an explanation about why, there is no "one column layout".
Since PowerPoint does not have such layout, Pandoc (thus Quarto) can't rely on that.
<img width="486" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/7aa97708-3e53-4a50-a63d-f6fb07f6bb6a">

**Comment by cderv** [comment]
In fact following,  https://github.com/quarto-dev/quarto-cli/discussions/5609#discussioncomment-5966184 I tested a bit more in depth to understand how pandoc works. The doc in Quarto about slide layout comes from Pandoc which does the attribution of Layout https://pandoc.org/MANUAL.html#powerpoint-layout-choice

When I render you document, and open the PPTX, I checked the layout used for the slides and it is Content with Caption.

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/0d91c7a3-7d37-47d8-a10e-9f59a91f64ab)

So I guess this is the expected layout but it is just not one column. 

You can check the different layout available by exporting the reference doc used by Pandoc in Quarto .
````
quarto pandoc --output reference.pptx --print-default-data-file reference.pptx
````

an then look for layouts
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/4b82450f-ed77-4acf-a354-642aba583668)


So it could be something with Pandoc itself not having the Slide layout you expect, or not assigning to the correct slide layout. 

You should be able to modify this reference doc and provide a modified one to tweak the output. For example, if you modify the Content with Caption layout in the reference doc, it will be used by Pandoc (I think it matched only on name)

Hope it helps

**Comment by NickCH-K** [comment]
Thank you for looking into this! This is very helpful.
